File: src/devices/tests/eval/serve-fidl/cpp/src/driver.cc
// Copyright 2026 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <lib/driver/component/cpp/driver_base2.h>
#include <lib/driver/component/cpp/driver_export2.h>
#include <lib/driver/logging/cpp/logger.h>

namespace {

class DriverServeFidl : public fdf::DriverBase2 {
 public:
  DriverServeFidl() : fdf::DriverBase2("driver_serve_fidl") {}

  zx::result<> Start(fdf::DriverContext context) override {
    fdf::info("DriverServeFidl started");

    return zx::ok();
  }
};

}  // namespace

FUCHSIA_DRIVER_EXPORT2(DriverServeFidl);

File: src/ui/input/drivers/goodix-gt92xx/rust/src/driver.rs
// Copyright 2026 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use fdf_component::{Driver, DriverContext, DriverError, Node, driver_register};
use log::info;

struct GoodixGt92xxDriver {
    _node: Node,
}

driver_register!(GoodixGt92xxDriver);

impl Driver for GoodixGt92xxDriver {
    const NAME: &str = "goodix-gt92xx";

    async fn start(mut context: DriverContext) -> Result<Self, DriverError> {
        info!("GoodixGt92xxDriver (Rust Skeleton) started!");
        let _node = context.take_node()?;
        Ok(Self { _node })
    }

    async fn stop(&self) {
        info!("GoodixGt92xxDriver (Rust Skeleton) stopped!");
    }
}

#[cfg(test)]
mod tests {
    #[test]
    fn test_placeholder() {}
}
